src/main/scala/Joint.scala

```diff
@@ -32,7 +32,73 @@ object Joint {
     val n = DenseVector[Double](normalVec._1, normalVec._2, normalVec._3)
     -(n dot w)/linalg.norm(n)
   }
+
+  /**
+   * Find the vector indicating dip direction of the joint plane. Global positive x-axis points North and
+   * z-axis oriented with positive upward. Positive y-axis will point west based on this orientation.
+   * @param normalVec Normal vector to the joint plane
+   * @return Dip direction of the plane, indicating direction of greatest increase in z. Return as vector (a, b, 0)
+   */
+  private def dipDirVector(normalVec: (Double, Double, Double)) = {
```

Comment:
I think it's a good general practice to annotate functions with their return types, even if the compiler can technically infer them. It helps make sure that the function does what you expect it to because now you'll get a compiler error if the return type you specified doesn't match what your code. So this would become
```scala
private def dipDirVector(normalVec: (Double, Double, Double)): DenseVector[Double] = {
```

You can change the `dipDir` and `dipAngle` functions similarly.